src/components/ImageSelector/ImageSelector.vue

```diff
@@ -45,57 +275,537 @@
+      clearTimeout(this.historyTimeout);
+      this.historyTimeout = undefined;
+      this.endDisplay = true;
+    },
+    relatedImages: function(images) {
+      this.endDisplay = false;
+      this.displayRelatedImages(images);
+      // The animations take 8 seconds so store history only at the end.
+      this.historyTimeout = setTimeout(() => {
+        this.$store.dispatch("insertHistory", {
+          decade: this.currentDecade,
+          index: this.currentIndex,
+          data: this.data[this.currentIndex].imagePaths.square,
+        });
+        this.endDisplay = true;
+      }, 8000);
+    },
+    secondRelatedImages: function(relatedImages) {
+      if (
+        this.relatedImagesPosition.filter((e) => e.hover === true).length > 0
+      ) {
+        this.displaySecondRelatedImages(relatedImages);
+        let newData;
+        let newIndex = -1;
+        // Load bundle of images related to the selected related image
+        this.secondRelatedImageTimeout.push(
+          setTimeout(() => {
+            this.couldLoad = false;
+            newData = this.getImagesByDecade(this.nextDecade);
+            if (newData === undefined) {
+              this.loadDataWithSkipId(this.nextDecade, this.nextId);
+            } else {
+              newIndex = newData.data.findIndex((e) => {
+                return e.id == this.nextId;
+              });
+              if (newIndex === -1) {
+                this.loadDataWithSkipId(this.nextDecade, this.nextId);
+              }
+            }
+          }, 2000)
+        );
+        // Setup state of new center image
+        this.secondRelatedImageTimeout.push(
+          setTimeout(() => {
+            this.couldLoad = true;
+            this.relatedImagesPosition = this.secondRelatedImagesPosition;
+            this.secondRelatedImagesPosition = [];
+            this.currentXPosition = this.defineLeftPositionCenterPage();
+            this.currentYPosition = this.defineTopPositionCenterPage();
+            this.carouselHover = true;
+            this.currentDecade = this.nextDecade;
+            if (newIndex !== -1) {
+              this.currentIndex = newIndex;
+              this.data = newData.data;
+            } else {
+              this.data = this.getImagesByDecade(this.currentDecade).data;
+              const id = this.data.findIndex((e) => {
+                return e.id == this.nextId;
+              });
+              this.currentIndex = id;
+            }
+
+            this.shouldRunCentralImageTransition = false;
+            window.scrollTo(this.currentXPosition, this.currentYPosition);
+            this.$store.dispatch("insertHistory", {
+              decade: this.currentDecade,
+              index: this.currentIndex,
+              data: this.data[this.currentIndex].imagePaths.square,
+            });
+          }, 8000)
+        );
+      }
+    },
+    // Watcher uses to manage the completion element
+    currentDecade: function() {
```

Comment:
probably you could use a computed property instead of watch

Reply:
Probably yes, but this watch manage some par tof the completion element, I don't know if it's a good idea to change it now. Maybe, that could be better during the refactoring of the component. What do you think?

Reply:
probably best to refactor later indeed